bloc-did-method.md

```diff
@@ -8,99 +8,165 @@ The `did:bloc` DID method allows groups of independent organizations to share cu
+  - A list of detached JWS signatures, from the stakeholders, on the config file
+  - A list of cached copies of the stakeholder config files
 
 ###### Consortium Endorsement Signatures
 Stakeholders endorse the contents of a consortium configuration file by signing the contents.
 
-Verification: The signatures array must be removed from the config file and the config file must be canonicalized (TODO: define how) before the signatures are verified against the file.
+First the file has appended data removed.
+- The signatures array is removed
+- The stakeholder config cache is removed
 
-##### Stakeholder Configurations
-Within `stakeholders/` are files for stakeholder configs.
+Verification: The signatures array must be removed from the config file and the config file must be canonicalized (as done by JWS<!--does jws even canonicalize or does it sign a particular string representation...-->) before the signatures are verified against the file.
 
-There must be at least one stakeholder config file here, for the consortium domain to be usable for resolving DIDs.
-
-Each of these files is named `[hash].json`, where `hash` is the SHA-256 hash of the contents of the file. This hash value is the value used in the consortium config to link to this stakeholder config.
+##### Stakeholder Config Files
+Each of these files is named `[domain].json`, where `[domain]` is the URL domain, owned by the stakeholder organization, where you can find the canonical copy of the stakeholder's configuration.
 
 A stakeholder configuration file contains:
 - The stakeholder's domain
+- The stakeholder's DID (did:bloc)
+  - TODO: also include a did:web DID doc? Same DID, same verkeys, used for bootstrapping.
 - Stakeholder custom configuration settings
 - The stakeholder's Sidetree endpoints
-- A detached JWS signature on the file, created by the stakeholder
-- a local hashlink to the previous version of this config file
+- the hash of the previous version of this config file
 
-TODO: Specify how the stakeholders are identified as the owners of the signing keys. For example, they could have DIDs, with the docs resolvable in the consortium (with some appropriate method of bootstrapping trust), or they could have x.509 certs.
+TODO: this should probably be in a secondary file, or stored in a way that's easy to detach without risk of screwing up the payload text
+- A detached JWS signature on the file, created by the stakeholder
 
 ##### History
 The `history/` directory contains historical consortium configs. Each such file is named `[hash].json`, where `[hash]` is the SHA-256 hash of the given file.
 
+TODO: instead of separate history files, have a complete history database file, as either an object or a concat JSON?
+
+TODO: consortium history may need to cache the stakeholder configs and verkeys - since these can be replaced on the stakeholder domain (or the stakeholder could disappear completely), with the associated consortium configs being subsequently unverifiable.
```

Comment:
suggest storing these as sidetree "DID-like" documents.

Also, a few lines above is the claim that this config is signed solely by the stakeholder who may no longer be part of the consortium at some point in time. How is trust in this custody chain maintained when the signature's owner is no longer part of the consortium? Does the consortium configuration's history come into play at this point?

Reply:
Are sidetree "DID-like" documents supported in the sidetree spec? Or would I need to specify the means for storing a document that isn't a DID doc?

I think for verification, removed stakeholders won't pose an issue - if their did docs remain in sidetree, the consortium config (or a historical config) will point to them with its list of stakeholders, so they can be retrieved for signature verification.

Reply:
> Are sidetree "DID-like" documents supported in the sidetree spec?

No. As per today's latest discussion, this would be a separate API that would likely reuse some sidetree components.

> I think for verification, removed stakeholders won't pose an issue - if their did docs remain in sidetree, the consortium config (or a historical config) will point to them with its list of stakeholders, so they can be retrieved for signature verification.

Maybe, but elsewhere in this doc is the claim:

```
Automatic bootstrapping relies on the assumption of trust in the owner of a web domain. In resolving a did:bloc DID, the resolver assumes the consortium domain in the DID is safe, and that the stakeholder domains within the consortium config are also safe and correct.
```

There's also a related TODO regarding did:web DIDs.

I think the solution would involve having N number of stakeholders attesting to all components that imbue trust in this method. We already have the stakeholders signing the consortium domain's config file. There should also be proof that N stakeholders attested to the existence of a stakeholder at some point in time. First thought would be for them to sign the stakeholder's config file, but that could disappear along with the stakeholder's domain at any given moment. This proof should be stored if not the sidetree/"did-like" layer, then perhaps inside the consortium domain's `.well-known/did-bloc`.

Reply:
There's a list of stakeholders within the consortium config, so as part of the payload it is signed by all stakeholders. It indicates the DID and domain of each stakeholder, so it provides proof that each stakeholder (which signs the config) attests to the existence of each other stakeholder (identified within the stakeholder list).